src/FunctionRef.h

```diff
@@ -0,0 +1,54 @@
+﻿
+#pragma once
+
+// Declared only so it can be partially specialized
+template <class Signature>
+class cFunctionRef;
+
+/** A light-weight, type-erased reference to a function object.
+Non-owning, be mindful of lifetimes. */
```

Comment:
Should be worth noting that this is almost the same as std::function, but the difference is ...